src/proto.c

```diff
@@ -683,14 +683,17 @@ int scan_text_result(struct MBuf *pkt, const char *tupdesc, ...)
 				int newlen;
 				if (strncmp(val, "\\x", 2) != 0) {
 					log_warning("invalid bytea value");
+					va_end(ap);
```

Comment:
To avoid errors like this in the future, let's use the `goto failed` pattern for this function.